parse.lisp

```diff
@@ -320,9 +328,10 @@
 	   (process-short-option c :negated t)))
 	((long-option-p arg)
 	 (process-long-option arg))
-	(t
-         (push arg *command-line-arguments*) ; put the first non-option back before we return.
-	 (return))))
+        ((string= "--" arg) (return)) ; Stop processing on "--" option.
+	(t (push arg hold)))) ; Save positional arguments before options.
+    (setf *command-line-arguments* ; Restore positional from  before options arguments.
+          (append (reverse hold) *command-line-arguments*))
```

Comment:
So far, command-line-arguments took the same stance as a shell or as sbcl in stopping processing of options at the first non-option argument. I believe it's sensible. To change this policy requires justification. How do other popular getopt variants in this or other languages do it? Do we want to make it an option to support both modes? This is another backward-incompatible change that is not to be lightly made.

Reply:
Looks like you're right and this is even a POSIX standard.  I had done this initially just because it was easier than updating some client packages, but I'll revert this change and update the client packages to be more POSIXy.